test/integration/scheduler_perf/scheduler_test.go

```diff
@@ -43,6 +46,16 @@ func TestScheduling(t *testing.T) {
 					informerFactory, tCtx := setupTestCase(t, tc, nil, nil)
 
 					runWorkload(tCtx, tc, w, informerFactory)
+
+					if tc.FeatureGates[features.SchedulerQueueingHints] {
+						if err = checkEmptyInFlightPods(); err != nil {
+							tCtx.Errorf("%s: %s", w.Name, err)
+						}
+					}
+
+					// Reset metrics to prevent metrics generated in current workload gets
+					// carried over to the next workload.
+					legacyregistry.Reset()
```

Comment:
if we are keeping this line, do we need the extra logic in `metricsCollector.init`?

Reply:
Yes, we want to measure the configured metrics only for a step with `collectMetrics` set to true. That's why we need to reset these metrics in `metricsCollector.init` (not to interfere the results with the previous steps). The `legacyregistry.Reset()` is used to reset all metrics between workloads as they should be isolated.